docs/beta/plugins/discord.md

```diff
@@ -1,5 +1,248 @@
+
+- **Message**:
+
+A Discord message, e.g. ["Welcome to the SourceCred community!..."](https://discordapp.com/channels/453243919774253079/715770410955964477/715776215528505385) {SB: this is a link to the welcome message in the SourceCred Discord #start-here channel}. A message node will be connected to its author (user node creating the message), emoji reactions to the message, and any references to the message {SB: is this true??}. 
+
+[INSERT SCREENSHOT MESSAGE]
+
+- **Channel**:
+
+A Discord channel, e.g. [#start-here](https://discord.gg/qadM9u) {SB: link needs checking. Got link by clicking 'Create Invite' on the channel, but getting 'invalid link' msg for some reason when I try it locally). A channel node will be connected to all messages in that channel. 
+
+Found this in the code and pasting here in case the plugin exposes these channel types. 
+
+```
+ * Discord Channels can be of various types, defined below.
+ * See: https://discordapp.com/developers/docs/resources/channel#channel-object-channel-types
+ */
+export type ChannelType =
+  | "GUILD_TEXT"
+  | "DM"
+  | "GUILD_VOICE"
+  | "GROUP_DM"
+  | "GUILD_CATEGORY"
+  | "GUILD_NEWS"
+  | "GUILD_STORE";
+```
+
+- **Reaction**:
+
+A Discord emoji reaction, e.g. [insert cute emoji reaction code]. When a user reacts to a message, a reaction node is created that is connected to the author of the reaction (user that reacted) and the message reacted to. 
+
+
+[INSERT SCREENSHOT REACTIONS TO MESSAGE]
+
+
+### Edges
+
+The Discord plugin creates the following kinds of edges (connections between nodes) 
+in the contribution graph:
+
+- **Authors**:
+
+An "authors" edge connects an author (user or bot) to a message they created. 
+
+- **References**: 
+
+{SB: below is just copied and pasted from teh Discourse draft, as I'm not sure references count in discord}
+
+```
+A references edge connects a topic or post to another referencable node (i.e. a node that corresponds to a specific url on Discourse ). {SB: taken from GitHub plugin, still correct? Just say referenceable nodes are topics/posts/users?}
+
+If the reference is pointing to a user, we call it a "mention", but from SourceCred's perspective it's the same kind of edge. {SB: taken from GitHub plugin, still correct?}
+
+References in Discourse can be either hyperlinks to referencable nodes, or quoted parts of other posts {SB: I'm kind of guessing here, would be good to get more detailed explanation, and perhaps include a illustrative example (e.g. a quote that links to both text and image if supported?). This brings up another questions I've had for a while: when you select and click the "Quote" button in the UI, it automatically creates a reply, presumably creating both a reply edge and reference edge. But what if you've already started a reply, and are selecting quotes from the UI from a post different than the one you started replying to? What if you're quoting multiple posts in the same reply?}
+```
+
+- **Reply to**:
```

Comment:
Doesn't exist in the current implementation

Reply:
Removed.

Reply:
Removed.